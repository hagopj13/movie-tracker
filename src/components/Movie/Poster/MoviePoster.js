// @flow
import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { makeStyles } from '@material-ui/core/styles';
import CardMedia from '@material-ui/core/CardMedia';
import ImageIcon from '@material-ui/icons/Image';

import configSelectors from 'store/config/config.selectors';

type Props = {
  posterPath?: string,
  fullPosterPath: string,
  height?: number,
  width?: number,
};

const useStyles = makeStyles((theme) => ({
  brokenMediaContainer: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: theme.palette.grey[200],
  },
}));

export const MoviePoster = (props: Props) => {
  const { posterPath, fullPosterPath, height, width } = props;

  const classes = useStyles();

  const isImageFound = posterPath && fullPosterPath;
  const imageStyle = {};
  if (height) {
    imageStyle.height = height;
  }
  if (width) {
    imageStyle.width = width;
  }

  return isImageFound ? (
    <CardMedia image={fullPosterPath} style={imageStyle} />
  ) : (
    <div className={classes.brokenMediaContainer} style={imageStyle}>
      <ImageIcon fontSize="large" />
    </div>
  );
};

MoviePoster.defaultProps = {
  posterPath: null,
  height: 0,
  width: 0,
};

const mapStateToProps = createStructuredSelector({
  fullPosterPath: (state, ownProps: Props) =>
    configSelectors.createPosterFullPathSelector(ownProps.posterPath)(state),
});

export default connect(mapStateToProps)(MoviePoster);
