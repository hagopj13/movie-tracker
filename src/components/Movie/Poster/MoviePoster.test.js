import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';
import CardMedia from '@material-ui/core/CardMedia';
import ImageIcon from '@material-ui/icons/Image';

import { MoviePoster } from './MoviePoster';

describe('MoviePoster component', () => {
  let shallow;
  let posterPath;
  let fullPosterPath;
  let wrapper;

  beforeAll(() => {
    shallow = createShallow();
  });

  beforeEach(() => {
    posterPath = 'pathToPosterImage.png';
    fullPosterPath = 'pathToFullPosterImage.png';
    const mockProps = {
      posterPath,
      fullPosterPath,
    };
    wrapper = shallow(<MoviePoster {...mockProps} />);
  });

  it('should render the MoviePoster component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render CardMedia if fullPosterPath is defined', () => {
    expect(wrapper.exists(CardMedia)).toBe(true);
    expect(wrapper.find(CardMedia).prop('image')).toBe(fullPosterPath);
  });

  it('should render a div containing ImageIcon as media if fullPosterPath is not defined', () => {
    const mockProps = {
      posterPath: null,
      fullPosterPath: null,
    };
    const newWrapper = shallow(<MoviePoster {...mockProps} />);
    expect(newWrapper.exists(CardMedia)).toBe(false);
    expect(newWrapper.exists(ImageIcon)).toBe(true);
  });
});
