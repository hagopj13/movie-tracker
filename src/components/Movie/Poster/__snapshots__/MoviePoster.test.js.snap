// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviePoster component should render the MoviePoster component correctly 1`] = `
<WithStyles(ForwardRef(CardMedia))
  image="pathToFullPosterImage.png"
  style={Object {}}
/>
`;
